### ISSUE TITLE
No type err when expected!

### ISSUE THREAD

**Issue by KDPRoss** [facebook/pyre-check#53]
This code clearly has a type error (and fails at runtime: `TypeError: unsupported operand type(s) for ** or pow(): 'str' and 'str'`):

```python
def foo(x : int) -> str:
    return str(x ** x)

x = foo('hi')
```

running `pyre check`, however, returns `No type errors found` ... Not sure how to find my `pyre` version ... I *just* installed it via `pip` 10.0.1 with Python 3.6.5. I get the same error when `x` isn't used within `foo`, which was the original code (which, of course, doesn't actually fail because `x` isn't *used*) that I was trying with `pyre`:

```python
def foo(x : int) -> str:
    return 'hi'

x = foo('hi')
```

**Comment by KDPRoss** [comment]
Adding `# pyre-strict` catches the error. But the original code doesn't seem to match any of the listed cases for which `pyre` will not throw errors in 'default mode'.

**Comment by shannonzhu** [comment]
Hi,
Just to clarify, you mean the second example *did* throw a parameter error when you passed 'hi' into foo, but not the first example? That seems a bit strange, because when Pyre is analyzing a function body, it only looks at the signatures of other functions being called and so the behavior should be the same in this case.

Could you help clarify what type error you meant by "I get the same error when `x` isn't used within `foo`", and which additional error was caught with `pyre-strict`? I think I may be misunderstanding the issue here.

**Comment by dark** [comment]
You can find your pyre version via `pip list | grep pyre-check`.

**Comment by KDPRoss** [comment]
Cheers, dark! It's 0.0.6

**Comment by shannonzhu** [comment]
This looks like an issue with errors on toplevel statements. We're looking into fixing this!

**Comment by dkgi** [comment]
@KDPRoss: I'm currently looking into fixing this. Reporting on top-level function has revealed quite a few bugs in our preprocessing. Will keep you posted.

**Comment by dark** [comment]
I believe this has been fixed long time ago, hence closing.
Feel free to reopen / file a new issue if a similar problem pops up again.
